# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build expat-build freetype-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.13; lt-2.4.2)
libtoolize -c -f; aclocal; autoheader; autoconf; automake -a -c
./configure --disable-static --with-xmldir=/etc/fonts --with-templatedir=/etc/fonts/conf.avail --with-default-fonts=/share/fonts --prefix=

# build
make
make fonts.conf

# install
install -d $DESTDIR/bin
install fc-cache/.libs/fc-cache $DESTDIR/bin
install fc-cat/.libs/fc-cat $DESTDIR/bin
install fc-list/.libs/fc-list $DESTDIR/bin
install fc-match/.libs/fc-match $DESTDIR/bin
install fc-pattern/.libs/fc-pattern $DESTDIR/bin
install fc-query/.libs/fc-query $DESTDIR/bin
install fc-scan/.libs/fc-scan $DESTDIR/bin
install fc-validate/.libs/fc-validate $DESTDIR/bin
install -d $DESTDIR/etc/fonts
install -m 644 fonts.conf $DESTDIR/etc/fonts
install -d $DESTDIR/etc/fonts/conf.d
install -m 644 conf.d/60-latin.conf $DESTDIR/etc/fonts/conf.d
install -d $DESTDIR/include/fontconfig
install -m 644 fontconfig/fcfreetype.h $DESTDIR/include/fontconfig
install -m 644 fontconfig/fcprivate.h $DESTDIR/include/fontconfig
install -m 644 fontconfig/fontconfig.h $DESTDIR/include/fontconfig
install -d $DESTDIR/lib
install src/.libs/libfontconfig.so.1 $DESTDIR/lib
ln -sf libfontconfig.so.1 $DESTDIR/lib/libfontconfig.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 fontconfig.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/fontconfig-build.pc << EOF
Name: fontconfig
Version: 2.11.1
Description: font configuration library
Requires: $REQUIRES

devel=\\
/include/fontconfig/fcfreetype.h \\
/include/fontconfig/fcprivate.h \\
/include/fontconfig/fontconfig.h \\
/lib/libfontconfig.so \\
/lib/pkgconfig/fontconfig.pc

exec=\\
/bin/fc-cache \\
/bin/fc-cat \\
/bin/fc-list \\
/bin/fc-match \\
/bin/fc-pattern \\
/bin/fc-query \\
/bin/fc-scan \\
/bin/fc-validate \\
/etc/fonts/fonts.conf \\
/etc/fonts/conf.d/60-latin.conf \\
/lib/libfontconfig.so.1
EOF

exit

# patch
--- fontconfig-2.11.1.orig/Makefile.am
+++ fontconfig-2.11.1/Makefile.am
@@ -21,6 +21,8 @@
 #  TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
 #  PERFORMANCE OF THIS SOFTWARE.
 
+AUTOMAKE_OPTIONS = -Wno-portability
+
 SUBDIRS=fontconfig fc-case fc-lang fc-glyphname src \
 	fc-cache fc-cat fc-list fc-match fc-pattern fc-query fc-scan \
 	fc-validate conf.d test
--- fontconfig-2.11.1.orig/configure.ac
+++ fontconfig-2.11.1/configure.ac
@@ -39,6 +39,8 @@
 
 dnl ==========================================================================
 
+AM_MAINTAINER_MODE
+
 AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
 
--- fontconfig-2.11.1.orig/fontconfig/fontconfig.h
+++ fontconfig-2.11.1/fontconfig/fontconfig.h
@@ -123,7 +123,8 @@
 #define FC_USER_CACHE_FILE	    ".fonts.cache-" FC_CACHE_VERSION
 
 /* Adjust outline rasterizer */
-#define FC_CHAR_WIDTH	    "charwidth"	/* Int */
+#define FC_CHARWIDTH	    "charwidth"	/* Int */
+#define FC_CHAR_WIDTH	    FC_CHARWIDTH
 #define FC_CHAR_HEIGHT	    "charheight"/* Int */
 #define FC_MATRIX	    "matrix"    /* FcMatrix */
 
--- fontconfig-2.11.1.orig/src/fcobjs.h
+++ fontconfig-2.11.1/src/fcobjs.h
@@ -51,7 +51,7 @@
 FC_OBJECT (RGBA,		FcTypeInteger,	NULL)
 FC_OBJECT (SCALE,		FcTypeDouble,	NULL)
 FC_OBJECT (MINSPACE,		FcTypeBool,	NULL)
-FC_OBJECT (CHAR_WIDTH,		FcTypeInteger,	NULL)
+FC_OBJECT (CHARWIDTH,		FcTypeInteger,	NULL)
 FC_OBJECT (CHAR_HEIGHT,		FcTypeInteger,	NULL)
 FC_OBJECT (MATRIX,		FcTypeMatrix,	NULL)
 FC_OBJECT (CHARSET,		FcTypeCharSet,	FcCompareCharSet)
--- fontconfig-2.11.1.orig/src/fcobjshash.gperf
+++ fontconfig-2.11.1/src/fcobjshash.gperf
@@ -44,7 +44,7 @@
 "rgba",FC_RGBA_OBJECT
 "scale",FC_SCALE_OBJECT
 "minspace",FC_MINSPACE_OBJECT
-"charwidth",FC_CHAR_WIDTH_OBJECT
+"charwidth",FC_CHARWIDTH_OBJECT
 "charheight",FC_CHAR_HEIGHT_OBJECT
 "matrix",FC_MATRIX_OBJECT
 "charset",FC_CHARSET_OBJECT
--- fontconfig-2.11.1.orig/src/fcobjshash.h
+++ fontconfig-2.11.1/src/fcobjshash.h
@@ -275,7 +275,7 @@
 #line 26 "fcobjshash.gperf"
       {(int)(long)&((struct FcObjectTypeNamePool_t *)0)->FcObjectTypeNamePool_str40,FC_WIDTH_OBJECT},
 #line 47 "fcobjshash.gperf"
-      {(int)(long)&((struct FcObjectTypeNamePool_t *)0)->FcObjectTypeNamePool_str41,FC_CHAR_WIDTH_OBJECT},
+      {(int)(long)&((struct FcObjectTypeNamePool_t *)0)->FcObjectTypeNamePool_str41,FC_CHARWIDTH_OBJECT},
 #line 48 "fcobjshash.gperf"
       {(int)(long)&((struct FcObjectTypeNamePool_t *)0)->FcObjectTypeNamePool_str42,FC_CHAR_HEIGHT_OBJECT},
 #line 44 "fcobjshash.gperf"

# source
https://freedesktop.org/software/fontconfig/release/fontconfig-2.11.1.tar.gz
